### PR TITLE
fix(react-drawer): regression with mountNode prop not available after refactoring

### LINKED ISSUE #29515: [Bug]: Regression - mountNode not available
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 19.87 GB / 31.60 GB
  Browsers:
    Edge: Chromium (117.0.2045.60)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/s/modest-alex-z8gh3x

You will notice that there is not `mountNode` option in the intelisense.

## Actual Behavior
Mount Node is no longer available on the Preview Drawer component.

You can see this when running the build command in your project with a `mountNode` on a `DrawerOverlay` component.
VS Code also complains that the property isn't available.

## Expected Behavior
The `mountNode` property is still available to use in development so that targeting the node can be done for stuff like custom styling or behaviors.

`9.34.2` (maybe `9.34.3`?) changed this by changing the behavior to internal with PR#29392.
The mount node should still be exposed so that customization is possible with the node.
This issue didn't exist in `9.34.1`.

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by marcosmoura**
## Previous Behavior
In the last release, the Drawer got on a huge code refactor and one property was missing after that (mountNode).

## New Behavior
Restores the property by synchronizing with the Dialog component props

## Related Issue(s)
- Fixes #29515

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/InfoButton-mount_base.html">18</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/InfoButton-mount.html">11</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/Avatar-mount_base.html">633</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/Avatar-mount.html">625</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/Button-mount_base.html">311</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/Button-mount.html">313</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/Field-mount_base.html">1130</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/Field-mount.html">1098</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/FluentProvider-mount_base.html">690</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/FluentProvider-mount.html">707</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/FluentProviderWithTheme-mount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/FluentProviderWithTheme-mount.html">83</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">65</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">64</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">82</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/InfoButton-mount_base.html">18</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/InfoButton-mount.html">11</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/MakeStyles-mount_base.html">859</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/MakeStyles-mount.html">884</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/Persona-mount_base.html">1744</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/Persona-mount.html">1655</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/SpinButton-mount_base.html">1400</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29524/perf-test-react-components/SpinButton-mount.html">1396</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              | `69.863 kB`<br />`20.148 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `207.62 kB`<br />`59.701 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `42.142 kB`<br />`13.97 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.503 kB`<br />`2.22 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5f6bdabdd9494d94872d354b286696840726aac4'>5f6bdabdd9494d94872d354b286696840726aac4</a></sub>
<div id="bundleSizeComment9423"/>

**Review by ValentinaKozlova — APPROVED** [review]

**Review by Hotell — APPROVED** [review]

File: packages/react-components/react-drawer/src/components/DrawerOverlay/DrawerOverlaySurface/DrawerOverlaySurface.types.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export type DrawerOverlaySurfaceSlots = DialogSurfaceSlots;
 /**
  * DrawerOverlaySurface Props
  */
-export type DrawerOverlaySurfaceProps = ComponentProps<DrawerOverlaySurfaceSlots>;
+export type DrawerOverlaySurfaceProps = DialogSurfaceProps;
```
Comment (Hotell) [inline packages/react-components/react-drawer/src/components/DrawerOverlay/DrawerOverlaySurface/DrawerOverlaySurface.types.ts:12]: can we add tests that would catch this regression whenever dialog changes its apis ?

Comment (marcosmoura) [reply packages/react-components/react-drawer/src/components/DrawerOverlay/DrawerOverlaySurface/DrawerOverlaySurface.types.ts:12]: I'll add this in a follow up PR as the scope of those tests are bigger than this PR.

**Pull request merged**